lib/model/query/entities.js

```diff
@@ -307,6 +316,8 @@ const _updateEntity = (dataset, entityData, submissionId, submissionDef, submiss
     if (conflict !== ConflictType.HARD) { // We don't want to downgrade conflict here
       conflict = conflictingProperties.length > 0 ? ConflictType.HARD : ConflictType.SOFT;
     }
+  } else if (createSubAsUpdate) {
+    conflict = ConflictType.SOFT;
```

Comment:
How about adding a test or an assertion in the existing test about how `conflict` is `'soft'`?